libs/barista-components/filter-field/src/filter-field-config.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { InjectionToken } from '@angular/core';
+import { createTagDataForFilterValues } from './filter-field-util';
+
+/** User defined configurations that can override some settings of the filter-field component */
+export interface DtFilterValuesConfig {
+  tagValuesParser: Function;
```

Comment:
Instead of `Function` let's use a better type here. Something like:
```suggestion
  tagValuesParser: (filterValues: DtFilterValue[], editable?: boolean, deletable?: boolean) => DtFilterFieldTagData | null;
```

Note:
We also need to change the types `_DtFilterValue` and `_DtFilterFieldTagData` to be public so the consumer can use it.

`_DtFilterValue` -> `DtFilterValue`
`_DtFilterFieldTagData` -> `DtFilterFieldTagData`

Reply:
Update: I do not see the need for the params `editable` & `deletable`. Because either the consumer sets these properties for all generated tag the same, so this can be set statically in the custom parser function or the values of these properties depend on the filter values itself which are already provided as a param.

Reply:
So i would suggest the type: `(filterValues: DtFilterValue[]) => DtFilterFieldTagData | null`